sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricName.java

```diff
@@ -38,10 +41,13 @@
   public abstract String name();
 
   public static MetricName named(String namespace, String name) {
+    checkArgument(!Strings.isNullOrEmpty(namespace), "Metric namespace must be non-empty");
+    checkArgument(!Strings.isNullOrEmpty(name), "Metric name must be non-empty");
     return new AutoValue_MetricName(namespace, name);
   }
 
   public static MetricName named(Class<?> namespace, String name) {
```

Comment:
Check that `namespace` is not null? We'll get a `NullPointerException` without a useful message otherwise.

Reply:
Done.